mathesar_ui/src/components/column/ColumnName.svelte

```diff
@@ -31,4 +31,12 @@
   $: icon = getColumnIconProps(column);
 </script>
 
-<NameWithIcon on:click {icon} {isLoading}>{column.name}</NameWithIcon>
+<div class="column-name">
+  <NameWithIcon on:click {icon} {isLoading}>{column.name}</NameWithIcon>
+</div>
+
+<style>
+  .column-name {
+    padding: 0 0.5rem;
+  }
+</style>
```

Comment:
ColumnName is a common used in several places throughout the applications. There are other components such as TableColumn and SelectableColumn which compose over it and are used in other places.

Please be mindful when modifying the dom structure and styling of such components. Here are some of the places which are affected by this change.

Record selector:
![Screenshot 2022-08-22 at 11 03 05 PM](https://user-images.githubusercontent.com/11850603/185956761-e3bb3e11-96c1-4f06-a323-8035a5416b52.png)

Data explorer column selection pane:
![Screenshot 2022-08-22 at 11 04 14 PM](https://user-images.githubusercontent.com/11850603/185956829-a1f12291-2ac6-45ea-b714-965242a9a116.png)

Data explorer result header:
![Screenshot 2022-08-22 at 11 19 09 PM](https://user-images.githubusercontent.com/11850603/185957460-77e15a64-6689-411e-8114-df688fbc74dc.png)

Notice the issues with uneven padding and text overflow. Another place where this component is used while creating an unique constraint.

Reply:
@pavish I've removed the extra HTML and restored the padding at the cell level. However, in data explorer header I see that the button element is used, which adds padding, but it is not being used for table header, so we end up with additional padding around the result table header.

Reply:
Button is the right element to use for headers that are actionable. We should also be using a button for the table since it has a click action. It's a regression from one of the recent PRs, I've added a commit https://github.com/centerofci/mathesar/pull/1543/commits/76e4fa7b45c7a3a7dedf8ae37e829fb0ba4884a3 to address that.

Notice that this commit removes padding from the SheetHeader and moves it to the table's header cell.

Reason: The components within `sheet/` are meant to be headless. You can notice the other components within `sheet/` to not contain any elements, and just have a slot. In some components like the SheetHeader, we need to handle events such as scroll so they contain a basic wrapper. These components should not modify the layout of their children i.e. should not contain properties like padding, margin etc.,